### ISSUE TITLE
install error ubuntu

### ISSUE THREAD

**Issue by fipoucat** [dtcenter/MET#2105]
I am facing an install issue on ubuntu pc, below error message from make.log

usr/bin/g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../../src/basic/vx_cal -I../../../src/basic/vx_config -I../../../src/basic/vx_log -I../../../src/basic/vx_math -I../../../src/basic/vx_util -I../../../src/libcode/vx_afm -I../../../src/libcode/vx_analysis_util -I../../../src/libcode/vx_color -I../../../src/libcode/vx_data2d -I../../../src/libcode/vx_data2d_factory -I../../../src/libcode/vx_data2d_grib -I../../../src/libcode/vx_data2d_grib2 -I../../../src/libcode/vx_data2d_nccf -I../../../src/libcode/vx_data2d_nc_met -I../../../src/libcode/vx_data2d_nc_pinterp -I../../../src/libcode/vx_data2d_python -I../../../src/libcode/vx_python3_utils -I../../../src/libcode/vx_geodesy -I../../../src/libcode/vx_gis -I../../../src/libcode/vx_gnomon -I../../../src/libcode/vx_grid -I../../../src/libcode/vx_gsl_prob -I../../../src/libcode/vx_nav -I../../../src/libcode/vx_nc_obs -I../../../src/libcode/vx_nc_util -I../../../src/libcode/vx_pb_util -I../../../src/libcode/vx_plot_util -I../../../src/libcode/vx_pointdata_python -I../../../src/libcode/vx_ps -I../../../src/libcode/vx_pxm -I../../../src/libcode/vx_render -I../../../src/libcode/vx_regrid -I../../../src/libcode/vx_shapedata -I../../../src/libcode/vx_solar -I../../../src/libcode/vx_statistics -I../../../src/libcode/vx_stat_out -I../../../src/libcode/vx_bool_calc -I../../../src/libcode/vx_summary -I../../../src/libcode/vx_time_series -I../../../src/libcode/vx_series_data -I../../../src/libcode/vx_tc_util -I/home/sarr/MET101/external_libs/include -fopenmp -I/home/sarr/MET101/external_libs/include -I/home/sarr/MET101/external_libs/include -I/home/sarr/MET101/external_libs/include -I../../basic/vx_log -I../../basic/vx_util -I/usr/include/python3.8 -I/home/sarr/MET101/external_libs/include -DBLOCK4 -DWITH_GRIB2 -DWITH_PYTHON -DMET_BASE="\"/home/sarr/MET101/share/met\""   -MT libvx_util_a-data_cube.o -MD -MP -MF .deps/libvx_util_a-data_cube.Tpo -c -o libvx_util_a-data_cube.o `test -f 'data_cube.cc' || echo './'`data_cube.cc
mv -f .deps/libvx_util_a-data_cube.Tpo .deps/libvx_util_a-data_cube.Po
/usr/bin/g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../../src/basic/vx_cal -I../../../src/basic/vx_config -I../../../src/basic/vx_log -I../../../src/basic/vx_math -I../../../src/basic/vx_util -I../../../src/libcode/vx_afm -I../../../src/libcode/vx_analysis_util -I../../../src/libcode/vx_color -I../../../src/libcode/vx_data2d -I../../../src/libcode/vx_data2d_factory -I../../../src/libcode/vx_data2d_grib -I../../../src/libcode/vx_data2d_grib2 -I../../../src/libcode/vx_data2d_nccf -I../../../src/libcode/vx_data2d_nc_met -I../../../src/libcode/vx_data2d_nc_pinterp -I../../../src/libcode/vx_data2d_python -I../../../src/libcode/vx_python3_utils -I../../../src/libcode/vx_geodesy -I../../../src/libcode/vx_gis -I../../../src/libcode/vx_gnomon -I../../../src/libcode/vx_grid -I../../../src/libcode/vx_gsl_prob -I../../../src/libcode/vx_nav -I../../../src/libcode/vx_nc_obs -I../../../src/libcode/vx_nc_util -I../../../src/libcode/vx_pb_util -I../../../src/libcode/vx_plot_util -I../../../src/libcode/vx_pointdata_python -I../../../src/libcode/vx_ps -I../../../src/libcode/vx_pxm -I../../../src/libcode/vx_render -I../../../src/libcode/vx_regrid -I../../../src/libcode/vx_shapedata -I../../../src/libcode/vx_solar -I../../../src/libcode/vx_statistics -I../../../src/libcode/vx_stat_out -I../../../src/libcode/vx_bool_calc -I../../../src/libcode/vx_summary -I../../../src/libcode/vx_time_series -I../../../src/libcode/vx_series_data -I../../../src/libcode/vx_tc_util -I/home/sarr/MET101/external_libs/include -fopenmp -I/home/sarr/MET101/external_libs/include -I/home/sarr/MET101/external_libs/include -I/home/sarr/MET101/external_libs/include -I../../basic/vx_log -I../../basic/vx_util -I/usr/include/python3.8 -I/home/sarr/MET101/external_libs/include -DBLOCK4 -DWITH_GRIB2 -DWITH_PYTHON -DMET_BASE="\"/home/sarr/MET101/share/met\""   -MT libvx_util_a-data_plane_util.o -MD -MP -MF .deps/libvx_util_a-data_plane_util.Tpo -c -o libvx_util_a-data_plane_util.o `test -f 'data_plane_util.cc' || echo './'`data_plane_util.cc
data_plane_util.cc: In function âvoid fractional_coverage(const DataPlane&, DataPlane&, int, GridTemplateFactory::GridTemplates, bool, SingleThresh, const DataPlane*, const DataPlane*, double)â:
data_plane_util.cc:262:36: error: âshapeâ not specified in enclosing âparallelâ
  262 |      GridTemplate* gt = gtf.buildGT(shape, width, wrap_lon);
      |                         ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
data_plane_util.cc:254:9: error: enclosing âparallelâ
  254 | #pragma omp parallel default(none)                 \
      |         ^~~
make[4]: *** [Makefile:1002: libvx_util_a-data_plane_util.o] Error 1
make[4]: Leaving directory '/home/sarr/MET101/met-10.1.0/src/basic/vx_util'
make[3]: *** [Makefile:369: all-recursive] Error 1
make[3]: Leaving directory '/home/sarr/MET101/met-10.1.0/src/basic'
make[2]: *** [Makefile:371: all-recursive] Error 1
make[2]: Leaving directory '/home/sarr/MET101/met-10.1.0/src'
make[1]: *** [Makefile:426: all-recursive] Error 1
make[1]: Leaving directory '/home/sarr/MET101/met-10.1.0'
make: *** [Makefile:367: all] Error 2

Hope to solve this to follow the Metplus training.

Thank you

**Comment by kgerheiser** [comment]
Yep, I'm getting this too. Came here to post this, and found your issue. `default(none)` is set, but shape isn't explicitly added to the `shared()` variables. You can fix this by adding `shape` to the list of `shared` variables. I'm assuming it should be shared. It doesn't look like it's being written to, but someone that knows the code better may be able to comment.

I'm guessing Intel doesn't complain about it. I'm using GCC 9.3.0.

Edit: Looking a little more closely shape is defined as ` const GridTemplateFactory::GridTemplates shape`, so it should definitely be shared.

**Comment by fipoucat** [comment]
On which file should I look at this?
I'm using GCC 9.4.0

Thanks

**Comment by kgerheiser** [comment]
data_plane_util.cc

```
--- a/src/basic/vx_util/data_plane_util.cc	2022-03-18 22:29:42.000000000 -0400
+++ b/src/basic/vx_util/data_plane_util.cc	2022-03-18 22:30:44.000000000 -0400
@@ -253,7 +253,7 @@
 
 #pragma omp parallel default(none)                 \
    shared(mlog, dp, frac_dp, width, wrap_lon, t)   \
-   shared(use_climo, cmn, csd, vld_t, bad)         \
+   shared(use_climo, cmn, csd, vld_t, bad, shape)  \
    private(x, y, n_vld, n_thr, gp, v)
    {
```

**Comment by fipoucat** [comment]
something strange on my side because when I change the file I still get the
same error. When I open it again the file is again as before the change.
Look very strange

**Comment by fipoucat** [comment]
maybe a write permission, I am changing it and will try again

**Comment by fipoucat** [comment]
I have problems with the file, when changed after compiling I have the same
error message, and when I open it appears without the change!
confusing for me

**Comment by kgerheiser** [comment]
That's an issue with your system, permissions, or editor. I was able to compile with the given change.
